public/app/features/alerting/unified/components/silences/MatchedSilencedRules.tsx

```diff
@@ -1,110 +1,81 @@
 import { AlertLabels } from '../AlertLabels';
 import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
-import { AlertStateTag } from '../rules/AlertStateTag';
 
-type MatchedRulesTableItemProps = DynamicTableItemProps<{
-  matchedInstance: Alert;
-}>;
-type MatchedRulesTableColumnProps = DynamicTableColumnProps<{ matchedInstance: Alert }>;
+import { AmAlertStateTag } from './AmAlertStateTag';
 
-export const MatchedSilencedRules = () => {
-  const [matchedAlertRules, setMatchedAlertRules] = useState<MatchedRulesTableItemProps[]>([]);
-  const formApi = useFormContext<SilenceFormFields>();
-  const dispatch = useDispatch();
-  const { watch } = formApi;
-  const matchers: MatcherFieldValue[] = watch('matchers');
+interface Props {
+  amSourceName: string;
+  matchers: Matcher[];
+}
+
+export const MatchedSilencedRules = ({ amSourceName, matchers }: Props) => {
+  const { useGetAlertmanagerAlertsQuery } = alertmanagerApi;
   const styles = useStyles2(getStyles);
   const columns = useColumns();
 
-  useEffect(() => {
-    dispatch(fetchAllPromAndRulerRulesAction());
-  }, [dispatch]);
-
-  const combinedNamespaces = useCombinedRuleNamespaces();
-  useDebounce(
-    () => {
-      const matchedInstances = combinedNamespaces.flatMap((namespace) => {
-        return namespace.groups.flatMap((group) => {
-          return group.rules
-            .map((combinedRule) => combinedRule.promRule)
-            .filter((rule): rule is AlertingRule => isAlertingRule(rule))
-            .flatMap((rule) => findAlertInstancesWithMatchers(rule.alerts ?? [], matchers));
-        });
-      });
-      setMatchedAlertRules(matchedInstances);
-    },
-    500,
-    [combinedNamespaces, matchers]
+  const { currentData: alerts = [] } = useGetAlertmanagerAlertsQuery(
+    { amSourceName, filter: { matchers } },
+    { skip: matchers.length === 0 }
   );
 
+  const tableItemAlerts = alerts.map<DynamicTableItemProps<AlertmanagerAlert>>((alert) => ({
+    id: alert.fingerprint,
+    data: alert,
+  }));
+
   return (
     <div>
       <h4 className={styles.title}>
         Affected alert instances
-        {matchedAlertRules.length > 0 ? (
-          <Badge className={styles.badge} color="blue" text={matchedAlertRules.length} />
+        {tableItemAlerts.length > 0 ? (
+          <Badge className={styles.badge} color="blue" text={tableItemAlerts.length} />
         ) : null}
       </h4>
       <div className={styles.table}>
         {matchers.every((matcher) => !matcher.value && !matcher.name) ? (
           <span>Add a valid matcher to see affected alerts</span>
         ) : (
-          <DynamicTable
-            items={matchedAlertRules}
-            isExpandable={false}
-            cols={columns}
-            pagination={{ itemsPerPage: 5 }}
-          />
+          <DynamicTable items={tableItemAlerts} isExpandable={false} cols={columns} pagination={{ itemsPerPage: 10 }} />
```

Comment:
Why was it changed from `5` to `10`?

Reply:
Based on my own observation that 5 is too low and uncomfortable when you have more alerts 😄